src/VisualStudio/Core/Test.Next/Remote/SnapshotSerializationTests.cs

```diff
@@ -679,16 +680,20 @@ void VerifyOptions(CompilationOptions originalOptions)
 
         private static SolutionAsset CloneAsset(ISerializerService serializer, SolutionAsset asset)
         {
+            var testSerializer = (TestSerializerService)serializer;
             using var stream = SerializableBytes.CreateWritableStream();
             using (var writer = new ObjectWriter(stream, leaveOpen: true))
             {
-                serializer.Serialize(asset.Value, writer, CancellationToken.None);
+                testSerializer.GetTestAccessor().Serialize(
+                    asset.Value, writer, forTesting: true);
```

Comment:
>forTesting [](http://example.com/codeflow?start=41&length=10)

Is forTesting ever passed in as false for this and CreateChecksum?

Reply:
no.  we never call GetTestAccessor and then pass false.  but i still want the parameter to be obvious so that we know that we're explicitly asking for differing behavior, and that's why we're going through the test accessor.  If the accessor just called the method, passing in true, it woudln't be clear here that we want different behavior and need the accessor for that.